### PR TITLE
fix: MD031/blanks-around-fences

### PR DESCRIPTION
**Pull request by nschonni**
- Fenced code blocks should be surrounded by blank lines
- Trim trailing leading spaces
- Space around headers
- Code fence languages

### CODE REVIEW

File: docs/framework/winforms/controls/serializing-collections-designerserializationvisibilityattribute.md [hunk 1]
```diff
@@ -1,136 +1,140 @@
...
-  
-5. Type several strings in the **String Collection Editor**. Separate them by pressing the ENTER key at the end of each string. Click **OK** when you are finished entering strings.  
-  
+> The strings you typed appear in the <xref:System.Windows.Forms.TextBox> of the `SerializationDemoControl`.
+
+## Serializing a Collection Property
+
+To test the serialization behavior of your control, you will place it on a form and change the contents of the collection with the **Collection Editor**. You can see the serialized collection state by looking at a special designer file, into which the **Windows Forms Designer** emits code.
+
+#### To serialize a collection
```
Comment (mairaw) [inline docs/framework/winforms/controls/serializing-collections-designerserializationvisibilityattribute.md]:
```suggestion
### To serialize a collection
```

File: docs/framework/winforms/controls/serializing-collections-designerserializationvisibilityattribute.md [hunk 2]
```diff
@@ -1,136 +1,140 @@
...
+
+## Prerequisites
+ In order to complete this walkthrough, you will need:
+
+- Sufficient permissions to be able to create and run Windows Forms application projects on the computer where Visual Studio is installed.
+
+## Creating a Control That Has a Serializable Collection
+ The first step is to create a control that has a serializable collection as a property. You can edit the contents of this collection using the **Collection Editor**, which you can access from the **Properties** window.
+
+#### To create a control with a serializable collection
```
Comment (mairaw) [inline docs/framework/winforms/controls/serializing-collections-designerserializationvisibilityattribute.md]:
```suggestion
### To create a control with a serializable collection
```

**Review by mairaw — COMMENTED** [review]
Changes look good @nschonni. Would you mind fixing the heading levels issue I found while reviewing this?

**Comment by nschonni** [comment]
@mairaw looks like the validation failures are related to the tab names, but that wasn't changed with this

**Comment by mairaw** [comment]
I know. We need to ask them to whitelist those values.

**Review by mairaw — APPROVED** [review]
Thanks!

**Pull request merged**